gpiozero/pins/data.py

```diff
@@ -422,13 +782,185 @@ class PiBoardInfo(namedtuple('PiBoardInfo', (
 
     .. attribute:: headers
 
-        A dictionary which maps header labels to dictionaries which map
-        physical pin numbers to :class:`PinInfo` tuples. For example, to obtain
-        information about pin 12 on header P1 you would query
-        ``headers['P1'][12]``.
+        A dictionary which maps header labels to :class:`HeaderInfo` tuples.
+        For example, to obtain information about header P1 you would query
+        ``headers['P1']``. To obtain information about pin 12 on header P1 you
+        would query ``headers['P1'].pins[12]``.
+
+        A rendered version of this data can be obtained by using the
+        :class:`PiBoardInfo` object in a format string::
+
+            from gpiozero import *
+            print('{0:headers}'.format(pi_info()))
+
+    .. attribute:: board
+
+        An ASCII art rendition of the board, primarily intended for console
+        pretty-print usage. A more usefully rendered version of this data can
+        be obtained by using the :class:`PiBoardInfo` object in a format
+        string. For example::
+
+            from gpiozero import *
+            print('{0:board}'.format(pi_info()))
 
     .. _system on a chip: https://en.wikipedia.org/wiki/System_on_a_chip
     """
+    __slots__ = () # workaround python issue #24931
+
+    @classmethod
+    def from_revision(cls, revision):
+        if revision & 0x800000:
+            # New-style revision, parse information from bit-pattern:
+            #
+            # MSB -----------------------> LSB
+            # uuuuuuuuFMMMCCCCPPPPTTTTTTTTRRRR
+            #
+            # uuuuuuuu - Unused
+            # F        - New flag (1=valid new-style revision, 0=old-style)
+            # MMM      - Memory size (0=256, 1=512, 2=1024)
+            # CCCC     - Manufacturer (0=Sony, 1=Egoman, 2=Embest)
+            # PPPP     - Processor (0=2835, 1=2836, 2=2837)
+            # TTTTTTTT - Type (0=A, 1=B, 2=A+, 3=B+, 4=2B, 5=Alpha (??), 6=CM, 8=3B, 9=Zero)
+            # RRRR     - Revision (0, 1, or 2)
```

Comment:
As mentioned earlier, this now goes "up to 3" for the newest PiZero, so perhaps change the comment to ` Revision (0, 1, 2, etc.)` ?